### PR TITLE
MM-40045 add checks for admin user in PermanentDeleteUser

### PR DESCRIPTION
**Pull request by titanventura**
#### Summary
Adds a team admin check to the permanentDeleteUser method

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/19997
Jira https://mattermost.atlassian.net/browse/MM-43447

### CODE REVIEW

**Comment by hanzei** [comment]
Some tests are failing. Could you please take a look?

**Comment by titanventura** [comment]
let me know if there is anything else @hanzei

**Comment by hmhealey** [comment]
@titanventura Sorry for the lack of responses. I think there's been some hesitation on reviewing this since it was assumed the tests were still broken and then when we went to re-run those, it appeared that the tests were broken by something in the enterprise code which is why people kept merging the latest version of master into the branch.

Despite all that, there is actually one remaining failure I'm seeing which is that your PR adds some new error messages to the code without adding them to `server/i18n/en.json`. Could you run `make i18n-extract` in the `server` folder and then update that file with an English message for both of those errors?

**Comment by titanventura** [comment]
Thanks for the review @hmhealey . And sure, will make the changes !

**Comment by hmhealey** [comment]
Thanks for updating that, but it looks like your editor has made some changes to the whitespace in that file which the i18n tooling is now complaining about. I forgot to mention it but you can run `make i18n-check` to locally check what the CI pipeline is checking

**Comment by titanventura** [comment]
reference: https://community.mattermost.com/core/pl/n361swdsmb8hurkrpzbbhqizsa

File: server/channels/api4/user_test.go [hunk 1]
```diff
@@ -2340,13 +2340,15 @@ func TestPermanentDeleteAllUsers(t *testing.T) {
 	})
 
 	t.Run("The endpoint should permanently delete all users", func(t *testing.T) {
-		// Basic user creates a team and a channel
-		team, appErr := th.App.CreateTeamWithUser(th.Context, &model.Team{
+		appErr := th.App.PermanentDeleteTeamId(th.Context, th.BasicTeam.Id)
+		require.Nil(t, appErr)
```
Comment (isacikgoz) [inline server/channels/api4/user_test.go:2344]: Why do we delete the basic team, can you elaborate that please?

File: server/channels/store/sqlstore/team_store.go [hunk 2]
```diff
@@ -1687,3 +1687,19 @@ func (s SqlTeamStore) GroupSyncedTeamCount() (int64, error) {
 
 	return count, nil
 }
+
+func (s SqlTeamStore) IsUserAdminOfTeam(userEmail string) (bool, error) {
+	builder := s.getQueryBuilder().Select("COUNT(*)").From("Teams").Where(sq.Eq{"Email": userEmail})
+	query, args, err := builder.ToSql()
+	if err != nil {
+		return false, errors.Wrap(err, "team_tosql")
+	}
```
Comment (isacikgoz) [inline server/channels/store/sqlstore/team_store.go:1696]: This request should've been checked by the permissions, and here the team e-mail is not reliable source. I would rather rely on permissions system.

File: server/channels/app/user.go [hunk 3]
```diff
@@ -1704,6 +1704,14 @@ func (a *App) PermanentDeleteUser(c request.CTX, user *model.User) *model.AppErr
 		c.Logger().Warn("You are deleting a user that is a system administrator.  You may need to set another account as the system administrator using the command line tools.", mlog.String("user_email", user.Email))
 	}
 
+	isUserAdminOfTeam, isUserAdminOfTeamErr := a.Srv().Store().Team().IsUserAdminOfTeam(user.Email)
```
Comment (isacikgoz) [inline server/channels/app/user.go:1707]: See my comment on the store method, this should be handled with the permissions. Also, I would try to tackle this issue within the API layer if possible.

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Sorry for the late review, this was way below my queue and not sure how it got there. Anyway, I'm requesting changes on design of this feature. Feel free to reach me out at https://community.mattermost.com for further clarification if that's necessary.

**Pull request closed without merging**